Chrysalis-Papilio/Soen490-Papilio-Website
# PAPILIO-101-task

## General
The current error message does not describe the situation accordingly. They should help the user understand what is happening to guide them towards the correct implementation of the forms. 
We should also make sure that error message happens onBlur meaning the user leaves the inputs and onSubmit meaning when the user submits the form. This will ensure the best UX possible.

_Issue number_: Papilio-101

## Task description
- create validation methods (required, match, hasLength)
- update useFormData to use validation forms
- create associated tests and update old test
- improve component usage of validation options

## Manual Test
Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)

1. start the backend
`npm run build && npm start`
2. start the website (it should be running to prove the npm conflict solution)
`npm start`
3. Click on get started
4. click on create without inputting a name in the business id field (you should see an error message)
5. Enter a valid name and proceed to the next page
6. Click next to the admin page
7. Click next, you should see error on all fields that match what is happening

## Testing
To run all the unit tests, run the following code: `npm test`
To make sure that the tests have reach the appropriate level of coverage, run the following code: `npm run coverage`
